### ISSUE TITLE
InvalidArgumentError (see above for traceback): Incompatible shapes: [2,1917] vs. [3,1]

### ISSUE THREAD

**Issue by anirudhkm** [tensorflow/models#5423]
I'm trying to develop a custom object detection model using online tutorials, and when I run the train.py script, I land up in the error _InvalidArgumentError (see above for traceback): Incompatible shapes: [2,1917] vs. [3,1]_.

Please, help in understanding the error and steps to fix it.

**Comment by anirudhkm** [comment]
Sorry for that, here are the answers to the above question

1. I am running the script from models/research/object-detection dir. I run the file train.py. Actually, the train.py file was in the directory _legacy_, so I copied it into models/research/object-detection.

2. Not any custom code, I have just followed a tutorial and tried to replicate the same. The tutorial that I follow is: https://www.youtube.com/watch?v=kq2Gjv_pPe8

3. I run on mac OS with Python 3.6 (Anaconda).

4. Installed tensorflow using the pip command.

5. Tensorflow version 1.10.1

6. Bazel version: Not sure about this as the tutorial didn't say anything about this.

7. CUDA/cuDNN: Same as above.

8. GPU model and memory: I think I'm just using CPU as I'm training on only 20 images.

9. Exact command used: python train.py --logtostderr --train_dir=training/ --pipeline_config_path=training/ssd_mobilenet_v1_pets.config

**Comment by anirudhkm** [comment]
Overview of the procedure followed:

1. Annotate image which will generate XML files.

2. Convert XML files to CSV file for both train and test data.

3. Using the CSV file, generate TF records.

4. Edit the config file, generate a .pbtxt file, once I have all these, I execute it as mentioned above.

**Comment by burrbank** [comment]
Currently running into a similar issue (following the same tutorial) using the legacy train.py. Have looked into a few things and havn't had much luck (reducing batch size, trying a couple different weights). Saw something about changing the bounding box coords from floats to ints but not sure if that makes any sense given that the box coords are normalised for the tfrecords.

**Comment by anirudhkm** [comment]
@burrbank  does your model have 6 classification labels?

**Comment by huseynlilkin** [comment]
I solved with [this](https://github.com/tensorflow/models/issues/5391#issuecomment-428176371) way.

**Comment by Shadowkm** [comment]
I still met this issue when using model_main.py

cant solved

**Comment by venkateshbabusekar** [comment]
even i am facing the same issue

**Comment by Shadowkm** [comment]
if need to solve it urgently, you can down version the tensorflow object-detection version
I am training by the previous version of object-detection of tensorflow from other github

**Comment by faheemirfan** [comment]
I am getting error even after using model_main.py..

It runs for some steps then give me this error

[[{{node cond_3/Detections_Left_Groundtruth_Right/3/_1699}} = _HostRecv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_220_cond_3/Detections_Left_Groundtruth_Right/3", tensor_type=DT_STRING, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

**Comment by rky0930** [comment]
I think this is a error associated with ssd_random_crop. 
In my case,  After removing ssd_random_crop option from pipeline.config, the object_detecion_api runs without error.

**Comment by engineer1982** [comment]
> 
> 
> I think this is a error associated with ssd_random_crop.
> In my case, After removing ssd_random_crop option from pipeline.config, the object_detecion_api runs without error.
> ssd_random_crop is very important feature to me, so i'm using previous version (ver. June 25, 2018).

Perfect @rky0930 ! that solved the problem. How did you find it?

And are you running the training in gpu? If yes, how long is each step taking? I am using a Tesla P100 and each step is taking 2.2 seconds

**Comment by rky0930** [comment]
Hi @engineer1982 
Haha...Fortunately, i remove the data_augmentation_options during copy and paste. 
And found out it ran well. 

GPUS: GTX1080(8Gb) x 2 
model: ssd_mobilenet_v2_coco 
batch_size: 64
the number of clones: 2 
1 step takes about 0.9~1 sec

```
# tail -f train.out
INFO:tensorflow:global step 25163: loss = 4.3094 (1.006 sec/step)
INFO:tensorflow:global step 25163: loss = 4.3094 (1.006 sec/step)
INFO:tensorflow:global step 25164: loss = 4.4951 (0.985 sec/step)
INFO:tensorflow:global step 25164: loss = 4.4951 (0.985 sec/step)
INFO:tensorflow:global step 25165: loss = 4.6927 (0.893 sec/step)
INFO:tensorflow:global step 25165: loss = 4.6927 (0.893 sec/step)
INFO:tensorflow:global step 25166: loss = 4.1873 (0.924 sec/step)
INFO:tensorflow:global step 25166: loss = 4.1873 (0.924 sec/step)
INFO:tensorflow:global step 25167: loss = 4.3566 (0.879 sec/step)
INFO:tensorflow:global step 25167: loss = 4.3566 (0.879 sec/step)
INFO:tensorflow:global step 25168: loss = 4.1848 (0.894 sec/step
```

**Comment by adnanobot** [comment]
> I think this is a error associated with ssd_random_crop.
> In my case, After removing ssd_random_crop option from pipeline.config, the object_detecion_api runs without error.
> ssd_random_crop is very important feature to me, so i'm using previous version (ver. June 25, 2018).

yea it worked for me as well after disabling data_augmentation_options function for ssd_random_crop.
thanks!!!

**Comment by GeneratorEX** [comment]
@adnanobot how can i remove data augmentation option? Can i disable it by removing that block of code?

**Comment by GeneratorEX** [comment]
@adnanobot do i need to comment it out also in the pipeline config file? Thank you

**Comment by GeneratorEX** [comment]
Thank you. Finally, the error seem to disappear. 

Did you manage to export the model? Im running export_inference_graph.py but it shows "Incomplete shape"

**Comment by adnanobot** [comment]
Yes, I did not face any problems afterwards.

**Comment by GeneratorEX** [comment]
@adnanobot did you encounter this error while running 'export_inference_graphy.py?

126 ops no flops stats due to incomplete shapes.
Parsing Inputs . . .
Incomplete shape.

**Comment by engineer1982** [comment]
> 
> 
> @adnanobot did you encounter this error while running 'export_inference_graphy.py?
> 
> 126 ops no flops stats due to incomplete shapes.
> Parsing Inputs . . .
> Incomplete shape.

In my case, I get those warnings but the train.py still exports the model and I am able to use it later.

**Comment by VladislavMz** [comment]
> i just simply commented the block like this (in the .config file, in my case it was ssd_mobilenet_v1_pets.config):
> 
> #data_augmentation_options {
> #ssd_random_crop {
> #}
> #}

Hello there! I spent a few days to make ssd_mobilenet_v1_coco work and finally I could! It works!!!

**Comment by GeneratorEX** [comment]
@engineer1982 did you encounter accuracy issues? My model didn't seem to work properly. It mistakenly detect all object it sees

**Comment by engineer1982** [comment]
@GeneratorEX , no.. the accuracy was fine

**Comment by GeneratorEX** [comment]
@engineer1982 My current training checkpoint is at 40k steps and the loss is not decreasing. Loss is stucked at 2 and sometimes decrease to 1. How can I fix this?

**Comment by engineer1982** [comment]
> 
> 
> @engineer1982 My current training checkpoint is at 40k steps and the loss is not decreasing. Loss is stucked at 2 and sometimes decrease to 1. How can I fix this?

Its very hard to tell how can you fix it.. It can be an error in generating the xml files or the model is not suitable for the task.. or other thing :(

If you are usin mobilenet, try to switch it to fast rcnn inception resnet v2. It is very accurate, but a little more slow to train and predict.

**Comment by gmt710** [comment]
> Hi @engineer1982
> Fortunately, i remove the data_augmentation_options during copy and paste.
> And found out it ran without error.
> 
> GPUS: GTX1080(8Gb) x 2
> model: ssd_mobilenet_v2_coco
> batch_size: 64
> the number of clones: 2
> 1 step takes about 0.9~1 sec
> 
> ```
> # tail -f train.out
> INFO:tensorflow:global step 25163: loss = 4.3094 (1.006 sec/step)
> INFO:tensorflow:global step 25163: loss = 4.3094 (1.006 sec/step)
> INFO:tensorflow:global step 25164: loss = 4.4951 (0.985 sec/step)
> INFO:tensorflow:global step 25164: loss = 4.4951 (0.985 sec/step)
> INFO:tensorflow:global step 25165: loss = 4.6927 (0.893 sec/step)
> INFO:tensorflow:global step 25165: loss = 4.6927 (0.893 sec/step)
> INFO:tensorflow:global step 25166: loss = 4.1873 (0.924 sec/step)
> INFO:tensorflow:global step 25166: loss = 4.1873 (0.924 sec/step)
> INFO:tensorflow:global step 25167: loss = 4.3566 (0.879 sec/step)
> INFO:tensorflow:global step 25167: loss = 4.3566 (0.879 sec/step)
> INFO:tensorflow:global step 25168: loss = 4.1848 (0.894 sec/step
> ```

It works.

**Comment by SteveIb** [comment]
> I think this is a error associated with ssd_random_crop.
> In my case, After removing ssd_random_crop option from pipeline.config, the object_detecion_api runs without error.
> ssd_random_crop is very important feature to me, so i'm using previous version (ver. June 25, 2018).

Hi,
In https://github.com/tensorflow/models/releases 
I didnt find 25th of June version, I only found 1 of June. Is that right version you pointed to?

**Comment by kumarvaibhav2003** [comment]
remove the data augmentation lines from the config file and it will run without error.

data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    ssd_random_crop {
    }
  }

**Comment by petchthanut** [comment]
I delete these lines in  **ssdlite_mobilenet_v2_coco.config** and **pipeline.config**

data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    ssd_random_crop {
    }
  }

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by amahendrakar** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
